### PR TITLE
Define product between a vector and a symmetric tensor.

### LINKED ISSUE #2413: vector*symmetric tensor doesn't work
This doesn't work:
```
  Tensor<1,dim> v;
  SymmetricTensor<2,dim> S;
  v*S*v;
```
whereas this works:
```
  Tensor<1,dim> v;
  SymmetricTensor<2,dim> S;
  v*(S*v);
```
Make the former happen as well.

(Or is this a bad idea, since we don't distinguish between row and column vectors this way?)

### PR DESCRIPTION
**Pull request by bangerth**
Also update some confusing pieces of documentation.

Fixes #2413.

### CODE REVIEW

File: include/deal.II/base/symmetric_tensor.h [hunk 1]
```diff
@@ -3067,16 +3067,8 @@ double_contract (SymmetricTensor<2,3,Number> &tmp,
...
- * first argument and the first index of the second argument. This function
- * therefore does the same as the corresponding <tt>contract</tt> function,
- * but returns the result as a return value, rather than writing it into the
- * reference given as the first argument to the <tt>contract</tt> function.
- *
- * Note that for the <tt>Tensor</tt> class, the multiplication operator only
- * performs a contraction over a single pair of indices. This is in contrast
- * to the multiplication operator for symmetric tensors, which does the double
- * contraction.
+ * Multiple a symmetric rank-2 tensor (i.e., a matrix) by a rank-1 tensor
```
Comment (davydden) [inline include/deal.II/base/symmetric_tensor.h]: Multiply?

File: include/deal.II/base/symmetric_tensor.h [hunk 2]
```diff
@@ -3095,6 +3087,23 @@ operator * (const SymmetricTensor<2,dim,Number> &src1,
 
 
 /**
+ * Multiple a rank-1 tensor (i.e., a vector) by a symmetric rank-2 tensor
```
Comment (davydden) [inline include/deal.II/base/symmetric_tensor.h]: same as above?

**Comment by davydden** [comment]
otherwise looks good to me :+1:

**Comment by bangerth** [comment]
Issues fixed -- I should not submit pull requests when I have to run to the gate to get onto the plane...

**Pull request merged**